### PR TITLE
feat: add call hash subcommand for script transactions

### PR DESCRIPTION
**Pull request by neutrinoks**
- To check pallet-move's storage for stored `Multisigs` in its `MultisigStorage`, we need the script transaction hash value
- Added a sub-command to calculate the call hash by a given script transaction

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,6 +16,7 @@ jsonrpsee = { version = "0.21", features = [ "http-client"] }
 tokio = { version = "1.36", features = ["rt", "time", "net"] }
 anyhow = "1.0"
 # Keep the version aligned with the clap version in the upstream repo https://github.com/move-language/move
+blake2 = "0.10"
```
Comment (Rqnsom) [inline Cargo.toml:18]: Should go above the comment above 😃

File: src/cmd/call_hash.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+use crate::cmd::{node::rpc::read_bytes, script_args::args::HexEncodedBytes};
+
+use anyhow::Result;
+use blake2::{Blake2s256, Digest};
+use clap::Parser;
+use std::path::PathBuf;
+
+/// Estimate gas for publishing modules.
```
Comment (Rqnsom) [inline src/cmd/call_hash.rs]: `/// Estimate gas for publishing modules.`
🕵️

File: src/cmd/node/rpc/mod.rs [hunk 3]
```diff
@@ -12,7 +12,7 @@ use std::fmt;
 use std::path::Path;
 
 /// Reads bytes from a file for the given path.
-fn read_bytes(file_path: &Path) -> Result<Vec<u8>> {
+pub(crate) fn read_bytes(file_path: &Path) -> Result<Vec<u8>> {
```
Comment (Rqnsom) [inline src/cmd/node/rpc/mod.rs]: We could move this `fn` somewhere else eventually, maybe into `src/mod.rs`

File: src/cmd/call_hash.rs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    pub fn execute(&self) -> Result<()> {
+        let script_tx = read_bytes(&self.script_transaction_path)?;
+
+        let mut hasher = Blake2s256::new();
+        hasher.update(&script_tx[..]);
+        let call_hash: [u8; 32] = hasher.finalize().into();
+
+        let call_hash_hex = HexEncodedBytes::from(call_hash);
+
+        println!("Call hash: {call_hash_hex}");
```
Comment (Rqnsom) [inline src/cmd/call_hash.rs:28]: Maybe an improvement for the future PR:
We could calculate and print the same line after someone executes `smove create-transaction [args...]`

**Review by Rqnsom — DISMISSED** [review]
LGTM

File: src/cmd/node/rpc/mod.rs (hunk 3 above)
Comment (neutrinoks) [reply src/cmd/node/rpc/mod.rs]: Done

File: src/cmd/call_hash.rs (hunk 2 above)
Comment (neutrinoks) [reply src/cmd/call_hash.rs]: Done

File: Cargo.toml (hunk 1 above)
Comment (neutrinoks) [reply Cargo.toml:18]: Done

**Review by Rqnsom — APPROVED** [review]

**Review by Rqnsom — APPROVED** [review]
Re-LGTM

**Pull request merged**